src/NGen.cpp

```diff
@@ -228,7 +228,15 @@ int main(int argc, char *argv[]) {
         double response = formulation_pair.second->get_response(time_step, 3600.0);
         //dump the output
         std::cout<<"\tCatchment "<<formulation_pair.first<<" contributing "<<response<<" m/s to "<<catchment_to_nexus[formulation_pair.first]<<std::endl;
-        catchment_outfiles[formulation_pair.first] << time_step <<", "<<response<<std::endl;
+        // If the timestep is 0, also write the header line to the file
+        // TODO: add command line or config option to have this be omitted
+        if (time_step == 0) {
+            // Append "Time Step" to header string provided by formulation, since we'll also add time step to output
+            std::string header_str = formulation_pair.second->get_output_header_line();
+            catchment_outfiles[formulation_pair.first] << "Time Step," << header_str <<std::endl;
```

Comment:
Really should pipe this output to the internal stream handler of the realization.

Reply:
Note this is stored via a StreamHandler in the HY_CatchmentArea as `output_stream` in a protected member.  Might need to move that and make the access a little simpler.

Reply:
Yeah, I agree this should be done, but I wasn't super familiar with the new StreamHandler type.   Also, the primary goal was to have the contents of the output be more complete and useful, so I was going to leave adjusting the overall output implementation for the driver to a separate issue.  I think your assessment was also to leave this until later, @hellkite500, but let me know if I have that wrong.